spec/frontdesk_spec.rb

```diff
@@ -0,0 +1,170 @@
+require "date"
+require_relative "spec_helper"
+
+describe "Frontdesk.new" do
+  before do
+    @frontdesk = Hotel::Frontdesk.new
+  end
+  it "creates and instance of Frontdesk" do
+    expect(@frontdesk).must_be_instance_of Hotel::Frontdesk
+  end
+
+  it "creates a list of 20 rooms at the hotel" do
+    expect(@frontdesk.rooms).must_be_instance_of Array
+    expect(@frontdesk.rooms.length).must_equal 20
+    expect(@frontdesk.rooms.first.number).must_equal 1
+    expect(@frontdesk.rooms.last.number).must_equal 20
+  end
+end
+
+describe "Frontdesk request_reservation" do
+  before do
+    @frontdesk = Hotel::Frontdesk.new
+    @reservation = Hotel::Reservation.new("Agatha Christie", "2019-05-20", 2)
+    @reservation_2 = Hotel::Reservation.new("Leni Zumas", "2019-05-23", 2)
+    @reservation_3 = Hotel::Reservation.new("Nnedi Okorafor", "2019-05-18", 2)
+    @reservation_4 = Hotel::Reservation.new("Justin Martinsen", "2019-05-17", 10)
+    @frontdesk.request_reservation(@reservation)
+    @frontdesk.request_reservation(@reservation_2)
+    @frontdesk.request_reservation(@reservation_3)
+    @frontdesk.request_reservation(@reservation_4)
+  end
+  it "adds reservation to reservation list" do
+    expect(@frontdesk.reservations[0]).must_be_instance_of Hotel::Reservation
+  end
+  it "assigns an available room to the reservation" do
+    expect(@reservation.room_num).must_equal 1
+  end
+  it "allows a new reservations checkin date to fall on a previous reservations checkout date" do
+    expect(@reservation_2.room_num).must_equal 1
+  end
+  it "allows a new reservations checkout date to fall on another reservations checkin date" do
+    expect(@reservation_3.room_num).must_equal 1
+  end
+  it "a new reservations can't completely overlap another" do
+    expect(@reservation_4.room_num).must_equal 2
+  end
+  it "raises and argument error if no rooms are available for that date" do
+    reservation_5 = Hotel::Reservation.new("Agatha Christie", "2019-05-20", 2)
+    expect {
+      19.times do
+        @frontdesk.request_reservation(reservation_5)
+      end
+    }.must_raise ArgumentError
+  end
+end
+
+describe "Frontdesk request_reservation with block_reference" do
+  before do
+    @frontdesk = Hotel::Frontdesk.new
+    @block_res = Hotel::Reservation.new("Octavia Butler", "2019-07-08", 3, block_reference: "SCIFI PARTY")
+    @frontdesk.request_block(@block_res, 5)
+    @reservation3 = Hotel::Reservation.new("Amy Martinsen", "2019-07-08", 3, block_reference: "SCIFI PARTY")
+    @frontdesk.request_reservation(@reservation3)
+    @reservation4 = Hotel::Reservation.new("Mollie Bullis", "2019-07-08", 2, block_reference: "SCIFI PARTY")
+  end
+  it "adjusts the reservation cost based on blocked status" do
```

Comment:
You should also test what happens when more reservations are requested than exist in the block.